### PR TITLE
Modest code motion in compile_fx

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #141656
* #141654
* __->__ #141574

Do code review with whitespace changes off. Check comments for what I changed.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: torch/_inductor/compile_fx.py [hunk 1]
```diff
@@ -1445,85 +1445,112 @@ def compile_fx(
...
+    Main entrypoint to a compile given FX graph.  Despite the fact that this
+    lives in :mod:`torch._inductor`, this function is responsible for calling
+    into AOT Autograd (and we will eventually get a callback to
+    ``inner_compile`` to perform actual compilation.  In other words, this
+    function orchestrates end-to-end compilation for the inductor backend when
+    you use :func:`torch.compile`.
+
+    NB: This function TAKES OWNERSHIP of the input ``model_`` and can potentially
+    mutate it!  Make a copy if you need to preserve the original GraphModule.
+    """
```
Comment (ezyang) [inline torch/_inductor/compile_fx.py:1458]: New docblock (and a few comments sprinkled here and there later)

File: torch/_inductor/compile_fx.py [hunk 2]
```diff
@@ -1688,10 +1722,6 @@ def bw_compiler(
                         boxed_forward_device_index=forward_device,
                     )
 
-        # TODO: can add logging before/after the call to create_aot_dispatcher_function
-        # in torch._functorch/aot_autograd.py::aot_module_simplified::aot_function_simplified::new_func
-        # once torchdynamo is merged into pytorch
```
Comment (ezyang) [inline torch/_inductor/compile_fx.py:1693]: torchdynamo is merged into pytorch so I assume this is obsolete

File: torch/_inductor/compile_fx.py [hunk 3]
```diff
@@ -1445,85 +1445,112 @@ def compile_fx(
...
-            with config.patch(config_patches):
-                return compile_fx(
-                    model_,
-                    example_inputs_,
-                    # need extra layer of patching as backwards is compiled out of scope
-                    inner_compile=config.patch(config_patches)(inner_compile),
-                    decompositions=decompositions,
-                )
+    """
+    Main entrypoint to a compile given FX graph.  Despite the fact that this
```
Comment (bobrenjc93) [inline torch/_inductor/compile_fx.py:1449]: nit: Main entry point for compiling a given FX graph

File: torch/_inductor/compile_fx.py [hunk 4]
```diff
@@ -1548,84 +1587,79 @@ def fw_compiler_base(
...
+                    # AOTAutograd will make those tensors being returned before the original
+                    # module's output.
+                    # To make things safe, we'll use original_output_start_index field
+                    # set by AOTAutograd to decide where the original module outputs start.
+                    orig_output_end_idx = (
+                        original_output_start_index + num_orig_model_outputs
                     )
-                else:
-                    original_output_start_index = 0
+                    # Sanity chec: we are about to splice out the "user" outputs from the full set
```
Comment (bobrenjc93) [inline torch/_inductor/compile_fx.py:1640]: s/chec/check

**Review by bobrenjc93 — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**